spec/adapters/heroku_spec.rb

```diff
@@ -0,0 +1,45 @@
+require "spec_helper"
+
+module Suspenders
+  module Adapters
+    RSpec.describe Heroku do
+      it "sets the heroku remotes" do
+        app_builder = double(app_name: app_name)
+        allow(app_builder).to receive(:append_file)
+
+        Heroku.new(app_builder).set_heroku_remotes
+
+        expect(app_builder).to have_received(:append_file).
+          with(anything, /heroku join --app #{app_name.dasherize}-production/)
+        expect(app_builder).to have_received(:append_file).
+          with(anything, /heroku join --app #{app_name.dasherize}-staging/)
+      end
+
+      it "setups the heroku specific gems" do
+        app_builder = double(app_name: app_name)
+        allow(app_builder).to receive(:inject_into_file)
+
+        Heroku.new(app_builder).setup_heroku_specific_gems
+
+        expect(app_builder).to have_received(:inject_into_file).
+          with("Gemfile", /rails_stdout_logging/, anything)
+      end
+
+      it "sets the heroku rails secrets" do
+        app_builder = double(app_name: app_name)
+        allow(app_builder).to receive(:run)
+
+        Heroku.new(app_builder).set_heroku_rails_secrets
+
+        expect(app_builder).to have_received(:run).
+          with(/SECRET_KEY_BASE.+staging/)
```

Comment:
It is, but I tried to test as much as it was being tested in the integration test. However, even with this I have my doubts, because aren't we coupling the test to much to the implementation, by having the entire command written down twice?